src/libutil/error.cc

```diff
@@ -334,7 +368,33 @@ std::ostream & showErrorInfo(std::ostream & out, const ErrorInfo & einfo, bool s
 
     bool frameOnly = false;
     if (!einfo.traces.empty()) {
+        std::set<Trace> tracesSeen;
+        std::vector<Trace> skippedTraces;
         size_t count = 0;
+
+        const auto printTrace = [&] (const Trace & trace) {
+            oss << "\n" << "… " << trace.hint.str() << "\n";
+
+            if (printPosMaybe(oss, ellipsisIndent, trace.pos))
+                count++;
+        };
+
+        const auto printSkippedTracesMaybe = [&] {
+            if (skippedTraces.size() > 0) {
+                // If we only skipped a few frames, print them out normally;
+                // messages like "1 duplicate traces omitted" aren't helpful.
+                if (skippedTraces.size() <= 5) {
+                    for (auto & trace : skippedTraces) {
+                        printTrace(trace);
+                    }
+                } else {
+                    oss << "\n" << ANSI_WARNING "(" << skippedTraces.size() << " duplicate traces omitted)" ANSI_NORMAL << "\n";
+                    tracesSeen.clear();
```

Comment:
(nit): Why do we care about that (and likely clearing `skippedTraces` below)?

Reply:
I've added a comment to explain this, but I'll copy it here:

Consider a mutually recursive stack trace with:
- 10 entries of A
- 10 entries of B
- 10 entries of A

If we don't clear `tracesSeen` here, we would print output like this:
- 1 entry of A
- (9 duplicate traces omitted)
- 1 entry of B
- (19 duplicate traces omitted)

This obscures the control flow, which went from A, to B, and back to A again.

In contrast, if we do clear `tracesSeen`, the output looks like this:
- 1 entry of A
- (9 duplicate traces omitted)
- 1 entry of B
- (9 duplicate traces omitted)
- 1 entry of A
- (9 duplicate traces omitted)

See: `tests/functional/lang/eval-fail-mutual-recursion.nix` for a test case exercising this property.